components/service/pocket/src/main/java/mozilla/components/service/pocket/stories/PocketRecommendationsRepository.kt

```diff
@@ -49,6 +59,32 @@ internal class PocketRecommendationsRepository(private val context: Context) {
         getAtomicStoriesFile().writeString { stories }
     }
 
+    suspend fun updateShownPocketRecommendedStories(updatedStories: List<PocketRecommendedStory>) {
+        synchronized(diskCacheLock) {
+            val currentStoriesArray = JSONObject(
+                getStoriesFile().inputStream().bufferedReader().use { it.readText() }
+            ).getJSONArray(JSON_KEY_ARRAY_STORIES)
+
+            val updatedStoriesArray = JSONArray()
+            lateinit var currentStory: JSONObject
+            val storiesLeftToUpdate = updatedStories.toMutableList()
+            for (index in 0 until currentStoriesArray.length()) {
+                currentStory = currentStoriesArray[index] as JSONObject
+
+                storiesLeftToUpdate.firstOrNull { currentStory.tryGetString(JSON_STORY_TITLE_KEY) == it.title }?.let {
```

Comment:
Is the title currently the only way how we can reference a story? (Would a database help with that too?) - Maybe the URL is guaranteed to be more unique than the title?